#Date: April 29, 2026
#Subject: [PATCH] Patch to fix compilation warning in libfyaml-master directory.
#
#Source: Uses code from the libfyaml-master and asprintf.c
#See NOTICE for license details.
#
#Signed-off-by: kanjoe24 <165808281+kanjoe24@users.noreply.github.com>
#---
#
#==========================================================================================
--- libfyaml-master/src/lib/fy-diag.c
+++ libfyaml-master/src/lib/fy-diag.c
@@ -18,6 +18,7 @@
 #include <ctype.h>
 
 #include <libfyaml.h>
+#include <asprintf.h>
 
 #include "fy-diag.h"
 
--- libfyaml-master/src/lib/fy-doc.c
+++ libfyaml-master/src/lib/fy-doc.c
@@ -18,6 +18,7 @@
 #include <limits.h>
 
 #include <libfyaml.h>
+#include <asprintf.h>
 
 #include "fy-parse.h"
 #include "fy-doc.h"
--- libfyaml-master/src/lib/fy-event.c
+++ libfyaml-master/src/lib/fy-event.c
@@ -19,6 +19,7 @@
 #include <stdarg.h>
 
 #include <libfyaml.h>
+#include <asprintf.h>
 
 #include "fy-parse.h"
 #include "fy-emit.h"
--- libfyaml-master/src/lib/fy-input.c
+++ libfyaml-master/src/lib/fy-input.c
@@ -27,6 +27,7 @@
 #endif
 
 #include <libfyaml.h>
+#include <asprintf.h>
 
 #include "fy-win32.h"
 #include "fy-parse.h"
--- libfyaml-master/src/lib/fy-token.c
+++ libfyaml-master/src/lib/fy-token.c
@@ -19,6 +19,7 @@
 #include <stdarg.h>
 
 #include <libfyaml.h>
+#include <asprintf.h>
 
 #include "fy-parse.h"
 
--- libfyaml-master/src/lib/fy-walk.c
+++ libfyaml-master/src/lib/fy-walk.c
@@ -20,6 +20,7 @@
 #include <float.h>
 
 #include <libfyaml.h>
+#include <asprintf.h>
 
 #include "fy-parse.h"
 #include "fy-doc.h"
--- libfyaml-master/src/lib/fy-parse.c
+++ libfyaml-master/src/lib/fy-parse.c
@@ -39,6 +39,7 @@
 static struct fy_eventp *
 fy_parser_event_resolve_hook_merge_key(struct fy_parser *fyp, struct fy_eventp *fyep);
 
+#if 0
 const char *fy_library_version(void)
 {
 #ifndef VERSION
@@ -48,6 +49,7 @@
 	return VERSION;
 #endif
 }
+#endif
 
 int fy_parse_input_append(struct fy_parser *fyp, const struct fy_input_cfg *fyic)
 {
@@ -4128,7 +4130,7 @@
 			if (c >= 0 && c <= 0x7f && (fy_utf8_low_ascii_flags[c] & F_SIMPLE_SCALAR)) {
 				size_t len, consumed;
 				const char *p, *s, *e;
-				int8_t cc;
+				int8_t cc = 0;
 				size_t run;
 
 				run = 0;
--- libfyaml-master/include/libfyaml/libfyaml-vlsize.h
+++ libfyaml-master/include/libfyaml/libfyaml-vlsize.h
@@ -810,13 +810,18 @@
 static inline const uint8_t *
 fy_decode_size_nocheck(const uint8_t *start, size_t *sizep)
 {
-	return fy_decode_size32_nocheck(start, sizep);
+	uint64_t sz;
+	const uint8_t *ret;
+
+	ret = fy_decode_size32_nocheck(start, &sz);
+	*sizep = (size_t)sz;
+	return ret;
 }
 
 static inline const uint8_t *
 fy_skip_size(const uint8_t *start, size_t bufsz)
 {
-	return fy_skip_size32(start, bufsz, &sz);
+	return fy_skip_size32(start, bufsz);
 }
 
 static inline const uint8_t *
